### PR TITLE
Add interactive UI documentation

### PR DESCRIPTION
**Pull request by ziad-saab**

### CODE REVIEW

**Comment by Montoya** [comment]
How do we get another preview published?

**Review by Montoya — APPROVED** [review]

**Comment by Montoya** [comment]
This is ready for review now

**Review by alexandratran — CHANGES_REQUESTED** [review]

File: snaps/features/custom-ui.md
Comment (alexandratran) [inline snaps/features/custom-ui.md:144]: Is the parameter `type` or `buttonType`?

File: snaps/features/custom-ui.md
Comment (alexandratran) [inline snaps/features/custom-ui.md:202]: Is the parameter `type` or `inputType`?

File: snaps/reference/snaps-api.md
Comment (alexandratran) [inline snaps/reference/snaps-api.md:13]: To clarify, the three new `Interface` methods do NOT require requesting permission in `initialPermissions`, but the rest of the methods do?

**Comment by alexandratran** [comment]
Made some edits for style and clarity. See my previous comments for outstanding questions.

File: snaps/features/custom-ui/index.md
Comment (alexandratran) [inline snaps/features/custom-ui/index.md:303]: I tried testing the `label` parameter but a label did not show up for me. The [example package](https://github.com/MetaMask/snaps/blob/main/packages/examples/packages/interactive-ui/src/index.ts) also doesn't seem to have a `label` example.

File: snaps/features/custom-ui.md
Comment (GuillaumeRx) [reply snaps/features/custom-ui.md]: Should be `buttonType`

File: snaps/features/custom-ui.md
Comment (GuillaumeRx) [reply snaps/features/custom-ui.md]: Should be `inputType`

File: snaps/reference/snaps-api.md
Comment (GuillaumeRx) [reply snaps/reference/snaps-api.md:13]: `snap_createInterface`, `snap_updateInterface`, `snap_getInterfaceState` and the `onUserInput` handler are all permitted meaning there's no permission needed to use those.

File: snaps/features/custom-ui/index.md
Comment (ziad-saab) [reply snaps/features/custom-ui/index.md:303]: cc. @GuillaumeRx

File: snaps/features/interactive-ui.md
Comment (ziad-saab) [reply snaps/features/interactive-ui.md]: thanks i'll add it back in

File: snaps/features/custom-ui/index.md
Comment (ziad-saab) [reply snaps/features/custom-ui/index.md:303]: The `label` property seems to still be in the code, I wonder if it's being used or not

https://github.com/MetaMask/snaps/blob/main/packages/snaps-sdk/src/ui/components/input.ts#L47

**Review by alexandratran — APPROVED** [review]
Additional minor edits and rearranged interactive UI methods in the Snaps API reference to clarify that those methods don't require requesting permission in the manifest file. LGTM

**Comment by ziad-saab** [comment]
Still waiting on an answer about `label` before merging

File: snaps/features/custom-ui/interactive-ui.md
Comment (Montoya) [inline snaps/features/custom-ui/interactive-ui.md:17]: Typo here, trailing ]

**Review by Montoya — DISMISSED** [review]

File: snaps/features/custom-ui/interactive-ui.md
Comment (ziad-saab) [reply snaps/features/custom-ui/interactive-ui.md]: fixed

**Review by GuillaumeRx — APPROVED** [review]

File: snaps/features/custom-ui/index.md
Comment (Montoya) [reply snaps/features/custom-ui/index.md:303]: This will be supported in 11.13, the docs are correct.

**Review by Montoya — APPROVED** [review]

**Pull request merged**
